### PR TITLE
Allow broader versions for psr/log

### PR DESCRIPTION
**Pull request by kissifrot**
Most other packages support 1/2/3.x versions

### CODE REVIEW

**Comment by neclimdul** [comment]
Did you check if the logger is compatible with the type hint changes in 3?

**Comment by tumispro** [comment]
This would be really useful, as it's not incompatible with a lot of Laravel applications.

**Comment by neclimdul** [comment]
Seems fine but waiting on confirmation of testing the breaking type hint changes in the PSR-3 v3 release.

**Comment by chumanfu** [comment]
I am also blocked by this. We are using Monolog in our app which requires psr/log ^2.0 so currently am unable to upgrade.

**Comment by chumanfu** [comment]
closes #159

**Review by chumanfu — APPROVED** [review]
👍🏻

**Comment by lucashodge** [comment]
Looks like this is all good, is it possible to get this merged? Thanks all.

**Comment by ryanwinchester** [comment]
@lucashodge @chumanfu did either of you test this branch to confirm it works?

**Comment by alex-volga** [comment]
+

**Comment by lucashodge** [comment]
@ryanwinchester @neclimdul sorry my previous comment wasn't clear then. All working fine on Drupal 10 for me with psr/log:3.0.0 . I am not sure if further more specific testing is required? 
I think a lot of people have just been forking this repo and applying the change but it would be good to use the OG repo again. Thanks for all your work.

**Comment by ahfeel** [comment]
6 months to get such a simple diff tested and merged, really ? ...

**Comment by jrebs** [comment]
> 6 months to get such a simple diff tested and merged, really ? ...

The author dropped the PR and then went completely silent. Never responded to any requests for confirmations, etc. It took about 6 months to get explicit verification that this had been tested against potential breaking combinations. Add to that the fact that this isn't a "something is broken in the current version" PR and instead is an "it would be nice if..." PR, yeah the urgency might not be really high for maintainers to do the leg work of getting a PR merged.

That being said, it's merged. Merry Christmas. 🎄🎁

**Comment by kissifrot** [comment]
Hello,

Sorry for the vanish, I had a car crash, thus I had different priorities than this PR 😅 

Thanks @jrebs for the merge and Merry Christmas too 🙂  🎄🎁

**Comment by ahfeel** [comment]
Thank you guys, hope you totally recovered @kissifrot !

Merry xmas !

**Pull request merged**
